### PR TITLE
Support for `cfg` attributes in host functions definitions

### PR DESCRIPTION
**Pull request by davxy**
As per title...

If I'm not terribly wrong, before this PR was not possible to define a host function gated by some `#[cfg(feature = ...)]`.  A similar technique has been used by `frame`.

---

Usage scenario: we require to early integrate and experiment with an host function that is still in "beta" phase as this one:

https://github.com/davxy/substrate/blob/5d2e2bbf17bb7f375d6d24bfe696be2c6e54aa2a/primitives/io/src/lib.rs#L1146-L1159

---

I'm not a rust macro wizard 🧙 so please carefully check

### CODE REVIEW

**Review by bkchr — APPROVED** [review]
A test with a feature that doesn't exist and thus, doesn't gets activated would be nice. (There is a testing crate for this stuff)

File: primitives/runtime-interface/proc-macro/src/runtime_interface/host_function_interface.rs [hunk 1]
```diff
@@ -96,6 +96,7 @@ fn generate_extern_host_function(
 		method.sig.ident,
 	);
 	let return_value = &method.sig.output;
+	let cfg_attrs = method.attrs.iter().filter(|a| a.path().is_ident("cfg"));
```
Comment (michalkucharczyk) [inline primitives/runtime-interface/proc-macro/src/runtime_interface/host_function_interface.rs:99]: nit: this filtering is repeated multiple time, maybe it is worth putting it to some shared util?

**Review by michalkucharczyk — APPROVED** [review]
Maybe it would be nice to have the example of usage in: primitives/runtime-interface/README.md.

**Review by michalkucharczyk — APPROVED** [review]
Maybe it would be nice to have the example of usage in: primitives/runtime-interface/README.md.

**Review by melekes — APPROVED** [review]
👍

File: primitives/runtime-interface/proc-macro/src/runtime_interface/host_function_interface.rs [hunk 2]
```diff
@@ -323,10 +330,19 @@ fn generate_host_function_implementation(
 		});
 	}
 
+	let cfg_attrs: Vec<_> = method
+		.attrs
+		.iter()
+		.filter(|a| a.path().is_ident("cfg"))
+		.map(|a| a.clone())
```
Comment (koute) [inline primitives/runtime-interface/proc-macro/src/runtime_interface/host_function_interface.rs:337]:
```suggestion
		.cloned()
```

(I *think* this should compile, but please check to be sure.)

**Comment by bkchr** [comment]
> It would be nice to have the example of usage in: primitives/runtime-interface/README.md.

Please add it to the crate docs as well (or only the crate docs), but never the README exclusively. Most of these readmes were generated from the crate docs.

**Pull request merged**